betty/config/factory.py

```diff
@@ -28,3 +36,48 @@ def new_for_configuration(
         """
         Create a new factory for the given configuration.
         """
+
+
+AnyConfigurableFactoryTarget: TypeAlias = (
+    ProjectTarget[_T] | ConfigurationDependentSelfFactory[_ConfigurationT]
+)
+"""
+A factory target for any service level that may be configurable.
+"""
+
+
+class _HumanFacingNewTargetError(FactoryError, HumanFacingException):
+    pass
+
+
+async def new_target(
+    target: AnyConfigurableFactoryTarget[_T, _ConfigurationT],
+    # @todo I think the problem is that mypy cannot map scalars to any of these types?
+    # @todo
+    # @todo
+    configuration: _ConfigurationT | Dump | Void = Void(),  # noqa B008
+    *,
+    factory: AnyFactory,
```

Comment:
Remove the factory argument. Instead, the function should return a factory target.